From 0a37645f4309b875f716471f8846d15282f08617 Mon Sep 17 00:00:00 2001
From: Avinash Kumar <avikumar@habana.ai>
Date: Wed, 23 Jun 2021 08:05:24 +0300
Subject: [PATCH] [SW-15543] gaudi: scl low i2c slave in debug mode

Before power-up Soc BMC assert HOST_GPIO to LOW level.
Preboot fw monitors HOST_GPIO as follows:
LOW level- Greco in debug mode. Fw will wait till
HOST_GPIO is asserted to HIGH.
HIGH level- FW will read and write to modules and
monitor HOST_GPIO level.

When user done reading/writing from Soc, he must assert
HOST_GPIO to high level. For gaudi host_gpio is GPIO[24].

Change-Id: Ifa9ec2cce78c51f70bda2d7b89cde5d46ee18f31
Signed-off-by: Avinash Kumar <avikumar@habana.ai>
---

diff --git a/arch/gaudi/common.c b/arch/gaudi/common.c
index 93eaa55..23c0e36 100644
--- a/arch/gaudi/common.c
+++ b/arch/gaudi/common.c
@@ -10,6 +10,7 @@
 #include <gaudi_blocks.h>
 #include <psoc_global_conf_regs.h>
 #include <psoc_efuse_regs.h>
+#include <psoc_gpio1_regs.h>
 #include <pcie_aux_regs.h>
 #include <pcie_core_regs.h>
 #include <pcie_wrap_regs.h>
@@ -56,6 +57,26 @@
 	((BIT(1) << CPU_CA53_CFG_ARM_RST_CONTROL_NCPUPORESET_SHIFT) |	\
 	 (BIT(1) << CPU_CA53_CFG_ARM_RST_CONTROL_NCORERESET_SHIFT))
 
+#define HOST_GPIO_I2C_DEBUG	24
+
+/* HOST_GPIO is set through mmPSOC_GPIO1 where each mmPSOC_GPIO serve 16 GPIOs.*/
+bool preboot_host_gpio_asserted(void)
+{
+	u32 reg, mask;
+
+	mask = BIT(HOST_GPIO_I2C_DEBUG % 16);
+
+	/* configure as input */
+	regmap_writel(mask, mmPSOC_GPIO1_OUTENCLR);
+
+	/* clear ALT function for gpio */
+	regmap_writel(mask, mmPSOC_GPIO1_ALTFUNCCLR);
+
+	reg  = regmap_readl(mmPSOC_GPIO1_DATA);
+
+	return reg & mask;
+}
+
 bool hl_is_debug_mode(enum hl_debug_hw_func func)
 {
 	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
